src/controller.ts

```diff
@@ -0,0 +1,132 @@
+//backend application framework for nodejs
+import {Express} from 'express';
+import { UserRepository } from './data';
+import { isSimpleUser, isIdentifiedUser } from './user';
+
+/**
+ * Broadcast-function 
+ * 
+ * @callback callbackBroadcast
+ * @param {string} tobeBroadcast The string, that is to be broadcast
+ */
+
+export class UserController {
+    //create a UserData-Object
+    private userRepo = new UserRepository();
+
+    /** 
+     * Creates an instance of UserController
+     * 
+     * @param {express.Express} app The express-app whose requests should be handled
+     * @param {callbackBroadcast} broadcast A function by which the controller can initiate a server broadcast
+     */
+    public constructor(private readonly app: Express, private broadcast: { (dataToBroadcast: string): void }) {
+        
+        //default get-route
+        this.app.get('/', (request, response, next) => {
+            console.log('Server: Received empty-path get-request!');
+            response.send("Hello World!");
+        })
+
+        //actual get-route, that provides the servers data
+        this.app.get('/getAllData', (request, response, next) => {
+            console.log('Server: Received proper get-request!');
+            response.send(JSON.stringify([...this.userRepo.getAllData()]));
+        })
+
+        //post request handler
+        //requires a simple ./user.User-object to be received
+        //saves received object's data in the users object, where it will receive an identifier and a joinedAt-Unix-timestamp
+        //updates all websocket clients automatically
+        this.app.post('/addUser', (request, response, next) => {
+            console.log('Server: Received post-request: %s', request.body);
+
+            //see if the data has the correct form
+            if (isSimpleUser(request.body)) {
```

Comment:
Checking the correct type of a request's body should be done in a related repository.